packages/react-admin/src/mui/modals/Modal.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import compose from 'recompose/compose';
+import shouldUpdate from 'recompose/shouldUpdate';
+
+import * as actions from './actions';
```

Comment:
Why import every possible actions ? That should be the responsibility of the user inside its render callback

Reply:
The render prop supplies these actions because they are redux connected actions. https://github.com/tkvw/admin-on-rest/blob/ce55ef916971e0bff19205cd9f3cfd1376742d00/packages/react-admin/src/mui/modals/Modal.js#L42
So Modal is nothing more than a bridge to these actions.

Reply:
Yes but why do we need this bridge. Shouldn't the consumer of the `Modal` just import what it needs by itself ?

Reply:
The entire purpose of `Modal` is: 
* to register a modal type (just a string identifier), which allows the Modal to be called by
* to abstract redux-promising-modals usage 

For a user to create a new modal is as simple as: 
```
<Modal type='myConfirm' render={(popModal)=>{
   <Dialog ...> 
}} /> 
```
This is much clearer than to create a connected Dialog and importing the appropriate `redux-promising-modals` actions (which requires more knowledge of redux-promising-modals). 

So it leads to cleaner code. 
But the user is free to implement it's own modal it's pretty straight forward. Basically just use mapstate and connect the appropriate actions.